sycl/include/sycl/ext/oneapi/experimental/invoke_simd.hpp

```diff
@@ -361,6 +363,20 @@ template <typename T>
 using strip_regcall_from_function_ptr_t =
     typename strip_regcall_from_function_ptr<T>::type;
 
+template <typename T> struct is_non_trivially_copyable_uniform {
+  static constexpr bool value =
+      is_uniform_type<T>::value &&
+      !std::is_trivially_copyable_v<typename unwrap_uniform<T>::type>;
```

Comment:
@rolandschulz , @Pennycook - is the trivially-copyable requirement applied to parameters ONLY ON DEVICE?

Let's say, sycl::local_accessor has different representation on HOST and DEVICE. It uses std::shared_ptr on HOST and simpler types on DEVICE. Such object is trivially_copyable on DEVICE

So, should this check be applied like this?
```
#ifdef __SYCL_DEVICE_ONLY__
check that invoke_simd arg is trivially copyable
#endif
```